cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/ExpressionHandler.kt

```diff
@@ -489,45 +410,14 @@ class ExpressionHandler(lang: CXXLanguageFrontend) :
     }
 
     private fun handleIdExpression(ctx: IASTIdExpression): DeclaredReferenceExpression {
-        val declaredReferenceExpression =
-            newDeclaredReferenceExpression(
-                ctx.name.toString(),
-                UnknownType.getUnknownType(language),
-                ctx.rawSignature
-            )
-        val proxy = expressionTypeProxy(ctx)
-        if (proxy is CPPClassInstance) {
-            // Handle Template Types separately
-            handleTemplateTypeOfDeclaredReferenceExpression(proxy, declaredReferenceExpression)
-        } else if (proxy !is TypeOfDependentExpression) {
-            declaredReferenceExpression.type =
-                TypeParser.createFrom(expressionTypeProxy(ctx).toString(), true, frontend)
-        }
-
-        /* this expression could actually be a field / member expression, but somehow CDT only recognizes them as a member expression if it has an explicit 'this'
-         */
+        // this expression could actually be a field / member expression, but somehow CDT only
+        // recognizes them as a member expression if it has an explicit 'this'
         // TODO: handle this? convert the declared reference expression into a member expression?
```

Comment:
Do we tackle this. Can this IdExpression even be a member expression?

Reply:
We sort of handle this later in the call resolver, so maybe we could remove the TODO or refer to the call resolver. The problem is that if you are within a class method, any reference to a name can either be a member expression or a simple reference to something outside of the class. For example

```
class Test {
  void test() {
    anotherTest(); // member (call) expression with implicit this
    yetAnotherTest(); // call expression to function outside of class
  }

  void anotherTest() {}
}

void yetAnotherTest().
```

In both cases we model it as a regular `CallExpression` with a `DeclaredReferenceExpression` and then later in the CXX specific call resolver, we handle it like it would have been a `MemberExpression` if we see that it refers to a method.